tests/table/table_io/test_csv.py

```diff
@@ -263,3 +263,91 @@ def test_csv_read_data_csv_no_header(use_native_downloader):
             read_options=TableReadOptions(column_names=["id", "data"]),
         )
         assert table.to_arrow() == expected.to_arrow(), f"Expected:\n{expected}\n\nReceived:\n{table}"
+
+
+@pytest.mark.parametrize("use_native_downloader", [True, False])
+def test_csv_read_data_csv_custom_quote(use_native_downloader):
+    with _csv_write_helper(
+            header=["\'id\'", "\'data\'"],
+            data=[
+                ["1", "\'aa\'"],
+                ["2", "aa"],
+                ["3", "aa"],
+            ]
+    ) as f:
+        storage_config = storage_config_from_use_native_downloader(use_native_downloader)
+
+        schema = Schema._from_field_name_and_types([("id", DataType.int64()), ("data", DataType.string())])
+        expected = Table.from_pydict(
+            {
+                "id": [1,2, 3],
+                "data": ['aa','aa','aa'],
+            }
+        )
+        table = table_io.read_csv(
+            f,
+            schema,
+            storage_config=storage_config,
+            csv_options=TableParseCSVOptions(quote='\''),
+        )
+
+        assert table.to_arrow() == expected.to_arrow(), f"Expected:\n{expected}\n\nReceived:\n{table}"
+
+# TODO this test still fails.
+@pytest.mark.parametrize("use_native_downloader", [False])
+def test_csv_read_data_csv_custom_escape(use_native_downloader):
+    with _csv_write_helper(
+            header=["id", "data"],
+            data=[
+                ["1", "\\\"aa\\\""],
+                ["2", "aa"],
+                ["3", "aa"],
+            ]
+    ) as f:
+        storage_config = storage_config_from_use_native_downloader(use_native_downloader)
+
+        schema = Schema._from_field_name_and_types([("id", DataType.int64()), ("data", DataType.string())])
+        expected = Table.from_pydict(
+            {
+                "\"id\"": [1,2, 3],
+                "\"data\"": ['\"aa\"','aa','aa'],
+            }
+        )
+        table = table_io.read_csv(
+            f,
+            schema,
+            storage_config=storage_config,
+            csv_options=TableParseCSVOptions(escape_char='\\'),
+        )
+
+        assert table.to_arrow() == expected.to_arrow(), f"Expected:\n{expected}\n\nReceived:\n{table}"
+
+#TODO Not testing use_native_downloader = False, as pyarrow does not support comments directly
+@pytest.mark.parametrize("use_native_downloader", [True])
```

Comment:
Skipping this should no longer be required now that you've added the row skipping UDF, right?

Reply:
Turns out, `pyarrow` only triggers the `invalid_row_handler` when there is a mismatch between the number of defined columns and the number of columns in the row.
So, this doesn't work. So, I'm stuck in a catch22. What do you think would be a solution? @clarkzinzow

Reply:
Given that `invalid_row_handler` is the canonical way to support skipping comments, I would then change the comment line in the CSV to contain a single "column", so that handler is triggered. That should be the common case, anyway.

Since `writer.writerows()` will fail in `_csv_write_helper()` when given the comment line, it might be better to bypass that helper here:
```python
    with tempfile.TemporaryDirectory() as directory_name:
        file = os.path.join(directory_name, "tempfile")
        with open(file, "w", newline="") as f:
            writer = csv.writer(f)
            writer.writerow(["id", "data"])
            writer.writerow(["1", "aa"])
            f.write("# comment line\n")
            writer.writerow(["3", "aa"])
        # ... rest of test
```